python/pyspark/sql/functions.py

```diff
@@ -2075,9 +2075,10 @@ class PandasUDFType(object):
 def udf(f=None, returnType=StringType()):
     """Creates a user defined function (UDF).
 
-    .. note:: The user-defined functions must be deterministic. Due to optimization,
+    .. note:: The user-defined functions are considered deterministic. Due to optimization,
         duplicate invocations may be eliminated or the function may even be invoked more times than
-        it is present in the query.
+        it is present in the query. If your function is not deterministic, call
+        `asNondeterministic`.
```

Comment:
Let's say this more explicitly like .. call `asNondeterministic()` in the user-defined function. It's partly because I think `UserDefinedFunction` is not documented in PySpark.

Reply:
yea, I think it is not clear here where to call `asNondeterministic` on. Maybe add a simple example.